### PR TITLE
Bugsnag 3 refactor

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

**Comment by loopj** [comment]
Nice! 

Standard James question - is there any way to make this new version of bugsnag-silex support both silex1 and silex2?

**Comment by GrahamCampbell** [comment]
Yes actually, with little pain! We can easily just add two different service providers since both silex 2 and 3 use the same symfony components.

**Comment by loopj** [comment]
Great! I think we should do that where possible.

**Comment by GrahamCampbell** [comment]
Ok, we support both versions now. :)

**Comment by kattrali** [comment]
This is legit 🎉  can you add an example project for silex 1 and 2 to the tree? I think its otherwise ready to go.

**Comment by GrahamCampbell** [comment]
Yeh, absolutely. That was the plan for Monday morning. There's already a link in our upcoming docs to a silex2 example project. :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+2.0.0
+-----
+
+-   Upgraded to the new PHP notifier
+-   Upgraded to Silex 2
```
Comment (kattrali) [inline CHANGELOG.md]: Since we support both versions now, we can say “Added support for”.

Comment (GrahamCampbell) [inline CHANGELOG.md]: Yeh :)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+test:
```
Comment (kattrali) [inline Makefile]: It’d be good to have a build target and probably test as the default:

``` make
default: test

build:
    composer install
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Contributing
+============
+
+-   [Fork](https://help.github.com/articles/fork-a-repo) the [notifier on github](https://github.com/bugsnag/bugsnag-laravel)
+-   Build and test your changes. Run the tests using [phpunit](https://phpunit.de) (vendored to `vendor/bin/phpunit`)
```
Comment (kattrali) [inline CONTRIBUTING.md]: If we’re using the makefile, then add instructions here for how to build and test.

File: README.md [hunk 4]
```diff
@@ -1,83 +1,39 @@
-Bugsnag Middleware for Silex
-==
+# Bugsnag Notifier for Silex
 
-The Bugsnag middleware for Silex integrates into Silex PHP applications.
+[![Build Status](https://img.shields.io/travis/bugsnag/bugsnag-silex/master.svg?style=flat-square)](https://travis-ci.org/bugsnag/bugsnag-silex)
```
Comment (kattrali) [inline README.md:2]: Add a docs link here, similar to the [badge on js](https://github.com/bugsnag/bugsnag-js/blob/master/README.md)

File: README.md [hunk 5]
```diff
@@ -1,83 +1,39 @@
...
-    ```json
-    {
-      "require": {
-        "bugsnag/bugsnag-silex": "^1.1"
-      }
-    }
-    ```
 
-2. If you haven't already, download and [install Composer][composer-download]:
+## Documentation
```
Comment (kattrali) [inline README.md]: Let’s swap this section out for the Features/Getting started format like [some of the others](https://github.com/bugsnag/bugsnag-python/blob/master/README.md#features)

File: src/Silex2ServiceProvider.php [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+use Pimple\ServiceProviderInterface;
+use Silex\Application;
+use Symfony\Component\HttpFoundation\Request;
+
+class Silex2ServiceProvider extends AbstractServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * Registers services on the container.
+     *
+     * @param \Pimple\Container $app
```
Comment (kattrali) [inline src/Silex2ServiceProvider.php:16]: “Pimple container”?

Comment (GrahamCampbell) [inline src/Silex2ServiceProvider.php:16]: Yeh, that's right. The silex 2 service provider is actually a pimple service provider. They moved it out to there.

File: Makefile (hunk 2 above)
Comment (GrahamCampbell) [inline Makefile]: We might need to sync up the other PHP libraries too.

File: src/Silex2ServiceProvider.php (hunk 6 above)
Comment (kattrali) [inline src/Silex2ServiceProvider.php:16]: Gotcha.

**Pull request merged**